Informatics II/002/Task 4/Yuirike Solution/Task4A1.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+/*
+I solved it before but had to get some help to figure out how to turn it.
+If anyone is good enough at code analysis, could you explain:
+
+Why n-1 seems to be such a crucial thing? I understand why it has to be in the recursive function but why also in the loop?
+
+Why does this work, i.e., why do I have to put the recursive call between for-loops?
+
+I tried to pull off this code with only the original array but that didn't work, is it because the original array 
+would contain other values? 
+*/
```

Comment:
Since those questions are asked discussed in the pr, they should be in the correctly working solution as comments.